tensorflow/lite/micro/micro_context.cc

```diff
@@ -0,0 +1,84 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/lite/micro/micro_context.h"
+
+#include <cstdarg>
+#include <cstddef>
+#include <cstdint>
+
+namespace tflite {
+MicroContext::MicroContext(MicroAllocator* allocator, const Model* model,
+                           MicroGraph* graph)
+    : allocator_(*allocator), model_(model), graph_(graph) {}
```

Comment:
No.   You can have pointer to references. 
1.  MicroGraph pointer needs to be returned for graph_ (GetGraph() for ResourceVariable)
2.  Model pointer is needed because SimpleMemoryAllocator's function takes pointer input.

Reply:
Able to do that for graph. I suspect that  model should be doable with some manuver (such as creating a empty model but without inovking new). In the spirit of avoiding this CL getting even bigger/complicated, will deferr that for later.

Reply:
Also, in the new change, you will see no separate accessor (const vs non-const) for  graph. All functions in MicroGraph are not const so there is no much meaning separation of different qualified accessor for now.